File: app/src/main/java/com/asafvaron/themoviedbtest/ui/movies/MoviesContract.java
package com.asafvaron.themoviedbtest.ui.movies;

import android.widget.ImageView;

import com.asafvaron.themoviedbtest.model.Movie;

import java.util.List;

/**
 * Created by asafvaron on 16/03/2017.
 */

public interface MoviesContract {

    /* View */
    interface View {

        void updateProgress(boolean visibility);

        void updateTitle(String title);

        void onSuccessMoviesLoaded(List<Movie> movies);

        void onFailedLoadMovies(String error);

        void onMovieClicked(ImageView imgView, Movie movie);

        void popInfoIfNeeded();
    }

    /* Model/Presenter communicator */
    interface Presenter {

        void loadMovies(String type);

    }
}

File: app/src/main/java/com/asafvaron/themoviedbtest/ui/movies/MoviesContract.kt
package com.asafvaron.themoviedbtest.ui.movies

import android.widget.ImageView

import com.asafvaron.themoviedbtest.model.Movie

/**
 * Created by asafvaron on 16/03/2017.
 */

interface MoviesContract {

    /* View */
    interface View {

        fun updateProgress(visibility: Boolean)

        fun updateTitle(title: String?)

        fun onSuccessMoviesLoaded(movies: List<Movie>?)

        fun onFailedLoadMovies(error: String?)

        fun onMovieClicked(imgView: ImageView?, movie: Movie?)

        fun popInfoIfNeeded()
    }

    /* Model/Presenter communicator */
    interface Presenter {

        fun loadMovies(type: String)
    }
}
